python/lsst/meas/base/plugins.py

```diff
@@ -68,6 +71,103 @@
+
+    def measure(self, measRecord, exposure):
+        # Promote the bounding box of the footprint to type D to ensure
+        # the ability to compare (they may be of mixed types I and D)
+        center = measRecord.getCentroid()
+        fpBbox = lsst.afw.geom.Box2D(measRecord.getFootprint().getBBox())
+        # check to insure that the center exists and that it is contained within the footprint
+        # catch bad centroiding
+        if not center or not fpBbox.contains(center):
+            raise Exception("The measRecord has no center, or the center is outside the footprint")
+
+        # Create an aperture and grow it by scale value defined in config to ensure there are enough
+        # pixles around the object to get decent statistics
+        aperture = lsst.afw.geom.ellipses.Ellipse(measRecord.getShape(), measRecord.getCentroid())
+        aperture.scale(self.config.scale)
+        foot = lsst.afw.detection.Footprint(aperture)
+        foot.clipTo(exposure.getBBox(lsst.afw.image.PARENT))
+        # Filter out any pixels which have mask bits set corresponding to the planes to be excluded
+        # (defined in config.mask)
+        maskedImage = exposure.getMaskedImage()
+        maskBits = maskedImage.getMask().getPlaneBitMask(self.config.mask)
+        logicalMask = numpy.logical_not(maskedImage.getMask().getArray() & maskBits)
+        # Compute the median variance value for each pixel not excluded by the mask and write the record
+        medVar = numpy.median(maskedImage.getVariance().getArray()[logicalMask])
+        measRecord.set(self.varValue, medVar)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.varFlag, True)
+
+class SingleFrameCountInputsConfig(SingleFramePluginConfig):
+    pass
+
+@register("base_CountInputs")
+class SingleFrameCountInputsPlugin(SingleFramePlugin):
+    """
+    Plugin to count how many input images contributed to each source. This information
+    is in the exposure's coaddInputs. Some Limitations:
+    * This is only for the pixel containing the center, not for all the pixels in the
+      footprint
+    * This does not account for any clipping in the coadd
+    """
+
+    ConfigClass = SingleFrameCountInputsConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.numberKey = schema.addField(name + '_value', type="I",
+                                         doc="Number of images contributing at center; not including any"
+                                             "clipping")
+        self.numberFlag = schema.addField(name + '_flag', type="Flag", doc="Set to True for fatal Failure")
+
+    def measure(self, measRecord, exposure):
+        if not exposure.getInfo() or not exposure.getInfo().getCoaddInputs():
+            raise lsst.pex.exceptions.RuntimeError("No coadd inputs defined")
```

Comment:
I'm not sure about this either, I ported the existing behavior from HSC and assumed the check was in there for some reason.

I would say we would prefer to fail rather than record number of inputs as 1 because presumably this plugin would be used assuming the exposure was a coadd. A failure would tell the user that the exposure was not a coadd vs. succeeding with all records being set to 1. This may introduce some unexpected behavior without user being aware.